share/util/shr_flux_mod.F90

```diff
@@ -362,13 +369,32 @@ SUBROUTINE shr_flux_atmOcn(nMax  ,zbot  ,ubot  ,vbot  ,thbot ,   &
         !------------------------------------------------------------
         !--- neutral coefficients, z/L = 0.0 ---
         stable = 0.5_R8 + sign(0.5_R8 , delt)
-        rdn    = sqrt(cdn(vmag))
+        if (wav_atm_coup .eq. 'twoway') then
+           if (z0wav(n) .lt. tiny) then 
+              ! z0wav == 0 for cold-start-WW3 situations   
+              cdn_wav = cdn_wave(loc_karman,zref,tiny)
+           else
+              cdn_wav = cdn_wave(loc_karman,zref,z0wav(n))
+           endif
+           rdn = sqrt(cdn_wav)
+           ! rdn calculated from Z0 will be constant
+        else
+          rdn    = sqrt(cdn(vmag))
+        endif
         rhn    = (1.0_R8-stable) * 0.0327_R8 + stable * 0.018_R8
                  !(1.0_R8-stable) * chxcdu + stable * chxcds
         ren    = 0.0346_R8 !cexcd
 
         !--- ustar, tstar, qstar ---
-        ustar = rdn * vmag
+        if (wav_atm_coup .eq. 'twoway') then
+           if (ustarwav(n) .lt. tiny) then 
+              ustar = ustarwav(n)+tiny 
```

Comment:
This is probably not very consequential, but this line should probably just be `ustar = tiny` to avoid introducing a discontinuity. If this line is left as-is, then if `ustarwav(n) == tiny`, then `ustar == tiny`, but if `ustarwav(n) == 0.9999 * tiny`, then `ustar == 1.9999*tiny`.